voxblox/include/voxblox/core/common.h

```diff
@@ -125,8 +125,10 @@ inline AnyIndex getGridIndexFromPoint(const Point& point,
 // function doesn't always compute the correct grid index for coordinates
 // near the grid cell boundaries.
 inline AnyIndex getGridIndexFromPoint(const Point& scaled_point) {
-  return AnyIndex(std::floor(scaled_point.x()), std::floor(scaled_point.y()),
-                  std::floor(scaled_point.z()));
+  const FloatingPoint kEpsilon = 1e-6;
+  return AnyIndex(std::floor(scaled_point.x() + kEpsilon),
+                  std::floor(scaled_point.y() + kEpsilon),
+                  std::floor(scaled_point.z() + kEpsilon));
```

Comment:
@ZacharyTaylor, yes this could happen. However, I think it is much more likely that the number is off by 1e-6 due to compound machine precision problems and actually belongs in the 0 grid, and not the -1 grid where it would be put if it is 1e-6.
@mfehr, the issue here and a bunch of other places was that coordinate -> grid -> coordinate -> grid wouldn't consistently map to the same grid points due to slight machine precision errors, it would get floored to the adjacent grid index. Since we use coordinate to grid conversions heavily for ray casting and many other operations, this was causing infinite loops in ray casting and sometimes accessing unallocated memory. Adding a tiny epsilon fixed this part of the issue in every case I could reproduce.